### PR TITLE
Use host cuda capability v2

### PR DESCRIPTION
**Pull request by abadams**
The buildbots seem uninterested in testing #3174 

Trying to open it as a new PR under a different branch.

### CODE REVIEW

**Comment by abadams** [comment]
It's the same commit hash, but now the buildbots are interested. It's alarming that they sometimes decide to ignore branches.

**Comment by steven-johnson** [comment]
I honestly don't know how to debug the buildbot weirdness -- surely there's a way monitor the buildbot master's pending queue (either in the GUI or cmd line), but I have never found it.

File: src/JITModule.cpp [hunk 1]
```diff
@@ -651,36 +661,57 @@ JITModule &make_module(llvm::Module *for_module, Target target,
         one_gpu.set_feature(Target::D3D12Compute, false);
         string module_name;
         switch (runtime_kind) {
-        case OpenCL:
+        case OpenCLDebug:
+            one_gpu.set_feature(Target::Debug);
+            module_name = "debug_";
+        case OpenCL: // fallthrough
```
Comment (steven-johnson) [inline src/JITModule.cpp]: This deliberately-fallthru stuff can cause a headache in some aggressive-warning variants of Clang, unless we annotate with `[[clang::fallthrough]]` or `[[gnu::fallthrough]]`. I'll add a suitable macro in a separate PR.

Comment (steven-johnson) [reply src/JITModule.cpp]: https://github.com/halide/Halide/pull/3211

**Comment by abadams** [comment]
Fallthroughs eliminated. They were providing minimal value anyway.

**Comment by steven-johnson** [comment]
LGTM

**Comment by steven-johnson** [comment]
This should probably be synced to master and re-tested if we want it to land soon. (Not gonna do that myself due to merge conflict.)

**Pull request closed without merging**